pkg/feature/cert.go

```diff
@@ -15,11 +16,46 @@ import (
 
 	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
 )
 
 var seededRand = rand.New(rand.NewSource(time.Now().UnixNano()))
 
+func (f *Feature) CreateSelfSignedCertificate(certificate v1.CertificateSpec, domain, namespace string) error {
```

Comment:
we could generate these certs using the rootca from k8s/ocp.

Reply:
I'm not sure, since the rootca is usually good for use inside the cluster, while this cert is for ingress traffic.